iib/workers/tasks/legacy.py

```diff
@@ -36,8 +36,13 @@ def export_legacy_packages(packages, request_id, rebuilt_index_image, cnr_token,
             _verify_package_info(package_dir, rebuilt_index_image)
             _zip_package(package_dir)
             _push_package_manifest(package_dir, cnr_token, organization)
-
-    set_request_state(request_id, 'in_progress', 'Back ported packages successfully pushed to OMPS')
+    payload = {
```

Comment:
This only gets called if `legacy_support_packages` is `True` in an `add` request and this is never called in an `rm` request, therefore, with your current PR, the request will never get to the `complete` state in many situations.

Reply:
You are right, I missed that. Let me modify my approach and update